### PR TITLE
add figher and levels and dice pools

### LINKED ISSUE #3: Implement `Fighter lvl 1`
- Basic support for the fighter class at level 1

### PR DESCRIPTION
**Pull request by Trisfald**
Not even a complete Fighter yet, but it's something.

I had to add many stuff: Die and Dices, Proficiency, Levels, etc. Hopefully a small `base` is starting to be born from the mess.

resolves #3 

After this PR we can start to work on actually `spawning` a character.

### CODE REVIEW

File: src/character/class/mod.rs [hunk 1]
```diff
@@ -21,4 +24,58 @@ impl From<&str> for ClassId {
...
+    #[test]
+    fn standard_proficiency_bonus() {
+        macro_rules! test_case {
+            ($( $lvl:expr, $expected:expr ),* $(,)?) => {
+                $(
+                    assert_eq!(proficiency_bonus_at_level(Level::new($lvl).unwrap()), $expected.into());
+                )*
+            }
+        }
+        test_case!(1, 2);
```
Comment (Trisfald) [inline src/character/class/mod.rs:60]: perhaps we could wrap the `macro` into a `macro` with ranges .. like `test_cases!(1..=4, 2)`
I have no idea how to do that

File: src/value.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    /// Sets a new maximum.
+    pub fn set_max(&mut self, max: T) {
+        self.max = Self::capped_value(max, max);
+        self.value = Self::capped_value(self.value, self.max);
+    }
+}
+
+impl<T: Default + Add + Copy + Ord + Add<T, Output = T>> PositiveBoundedValue<T> {
```
Comment (Trisfald) [inline src/value.rs:53]: `rust`'d

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: features.md [hunk 3]
```diff
@@ -220,8 +220,8 @@
     - [ ] Nature's Sanctuary
   - [ ] User Defined Druid Circles
 - [ ] Fighter
-  - [ ] Hit Points and Proficiency Bonus
-    - [ ] 1st Level
+  - [x] Hit Points and Proficiency Bonus
+    - [x] 1st Level
```
[the remark this answers is not in the document]
Comment (Trisfald) [reply features.md:224]: it works on my machine

File: src/ability.rs [hunk 4]
```diff
@@ -62,9 +62,11 @@ pub const CHARISMA: AbilityId = AbilityId(5);
 /// Number of core abilities.
 pub const RESERVED_ABILITIES: u8 = 6;
 
-const ABILITY_SCORE_MIN: u8 = 1;
+/// Minimum ability score.
+pub const ABILITY_SCORE_MIN: u8 = 1;
```
[the remark this answers is not in the document]
Comment (Trisfald) [reply src/ability.rs:66]: I'm not sure. It's kind of private, but maybe the user can use this information? Hm or not.

File: src/ability.rs [hunk 5]
```diff
@@ -99,7 +101,7 @@ impl AbilityScore {
         }
     }
 
-    /// Creates a new `AbilityScore` capped between the min and max value.
+    /// Constructs a new `AbilityScore` capped between the min and max value.
     pub fn capped(value: u8) -> Self {
         if value > ABILITY_SCORE_MAX {
```
[the remark this answers is not in the document]
Comment (Trisfald) [reply src/ability.rs:106]: almost like a macro

File: src/util.rs [hunk 6]
```diff
@@ -31,6 +31,10 @@ impl PackageVersion {
     }
 }
 
+pub(crate) fn is_value_valid<T: PartialOrd>(value: T, min: T, max: T) -> bool {
+    value >= min && value <= max
```
[the remark this answers is not in the document]
Comment (Trisfald) [reply src/util.rs:35]: debug_assert or assert?

File: src/ability.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply src/ability.rs:66]: I'd say he shouldn't, we should strive to encapsulate what we can. And later with shame accept our fate and decapsulate. :<

File: src/util.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply src/util.rs:35]: `assert`. If it occurs, the logic is flawed and there is no point in continuing. If it doesn't, the cost of a simple check is negligible.

File: src/ability.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Trisfald) [reply src/ability.rs]: Haha yes. They can read the manual to know the minimum.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**